packages/access/access-control/src/storage.rs

```diff
@@ -406,6 +406,39 @@ pub fn set_role_admin(e: &Env, role: &Symbol, admin_role: &Symbol) {
     set_role_admin_no_auth(e, role, admin_role);
 }
 
+/// Allows the current admin to renounce their role, making the contract
+/// permanently admin-less. This is useful for decentralization purposes or when
+/// the admin role is no longer needed. Once the admin is renounced, it cannot
+/// be reinstated.
+///
+/// # Arguments
+///
+/// * `e` - Access to Soroban environment.
+///
+/// # Errors
+///
+/// * [`AccessControlError::AdminNotSet`] - If no admin account is set.
+///
+/// # Events
+///
+/// * topics - `["admin_renounced", admin: Address]`
+/// * data - `[]`
+///
+/// # Notes
+///
+/// * Authorization for the current admin is required.
+pub fn renounce_admin(e: &Env) {
+    // Get the current admin and require authorization
+    let admin = get_admin(e);
```

Comment:
shouldn't we use `enforce_admin_auth()` as in the https://github.com/OpenZeppelin/stellar-contracts/pull/315/

Reply:
exactly! I tackled that issue later than this one. Trying to address each issue on its own PR. There will be lot of conflicts in Access Control in general :)